File: ruby/packages/var-runner/lib/oselvar/var/runner/baseline_store.rb
# frozen_string_literal: true

module Oselvar
  module Var
    module Runner
      # The filesystem BaselineStore: the committed drift baseline lives at the
      # project root as var.lock.json. The core owns the format; this adapter
      # only moves raw text. Port of baseline_store.py.
      class FileBaselineStore
        def initialize(root)
          @path = File.join(root.to_s, 'var.lock.json')
        end

        def read
          File.exist?(@path) ? File.read(@path, encoding: 'UTF-8') : nil
        end

        def write(contents)
          File.write(@path, contents)
        end
      end

      module_function

      def create_file_baseline_store(root)
        FileBaselineStore.new(root)
      end
    end
  end
end

File: typescript/packages/var-runner/src/baseline-store.ts
import { existsSync, mkdirSync, readFileSync, writeFileSync } from 'node:fs'
import { dirname, join } from 'node:path'
import type { BaselineStore } from '@varar/core'

// The committed drift baseline lives at the project root as varar.lock.json.
export function varLockPath(cwd: string): string {
  return join(cwd, 'varar.lock.json')
}

// The Node BaselineStore: varar.lock.json on disk. The core owns the format;
// this adapter only reads and writes the raw text.
export function createFileBaselineStore(cwd: string): BaselineStore {
  const path = varLockPath(cwd)
  return {
    read: () => (existsSync(path) ? readFileSync(path, 'utf8') : null),
    write: (contents: string) => {
      mkdirSync(dirname(path), { recursive: true })
      writeFileSync(path, contents)
    },
  }
}
